core/templates/dev/head/domain/exploration/StatesObjectFactory.ts

```diff
@@ -47,7 +47,7 @@ export class States {
       newStateName);
   }
   setState(stateName: string, stateData: any): void {
-    this._states[stateName] = Object.assign({}, stateData);
+    this._states[stateName].copy(stateData);
```

Comment:
Could you explain this change?

Reply:
`Object.assign(..)` returns an object with the content of stateData but since this._states is a collection of State objects we should use the copy method defined in the StateObjectFactory. This makes sure that `this._states[stateName]` remains a State object.

Reply:
Ah ok. Please add a detailed comment in the code (explaining what you just said), for future reference.

Reply:
Done.